parser.go

```diff
@@ -1275,151 +1273,7 @@ astFieldsLoop:
 		} else if isGoTypeOASType(typeAsString) {
 			fieldSchema.Type = goTypesOASTypes[typeAsString]
 		}
-
```

Comment:
realizing that this is difficult to review, a lot of it is just refactoring (no behavioral changes)